integration/base/src/main/java/com/inrupt/client/integration/base/AccessGrantScenarios.java

```diff
@@ -581,6 +613,69 @@ void accessGrantCreateNonRdfTest(final Session session) throws IOException {
         assertDoesNotThrow(accessGrantClient.delete(grant).toCompletableFuture()::join);
     }
 
+    private static void prepareACPofResource(final SolidSyncClient authClient, final URI resourceURI) {
+
+        final IRI acpAllOf = rdf.createIRI(ACP.allOf.toString());
+        final IRI acpVc = rdf.createIRI(ACP.vc.toString());
+        final IRI acpAllow = rdf.createIRI(ACP.allow.toString());
+        final IRI acpApply = rdf.createIRI(ACP.apply.toString());
+        final IRI acpAccessControl = rdf.createIRI(ACP.accessControl.toString());
+        final IRI aclRead = rdf.createIRI(ACL.Read.toString());
+        final IRI aclWrite = rdf.createIRI(ACL.Write.toString());
+
+        // find the acl Link in the header of the resource
+        final Request req = Request.newBuilder(resourceURI)
+                .HEAD()
+                .build();
+        final Response<Void> res = authClient.send(req, Response.BodyHandlers.discarding());
+        final Headers.Link acrLink = res.headers().allValues("Link").stream()
```

Comment:
I would have thought that our [Metadata](https://docs.inrupt.com/developer-tools/api/java/inrupt-client/latest/com/inrupt/client/solid/Metadata.html) class already had a mechanism for extracting this URL, but it appears that it doesn't. That would be a nice feature of the `Metadata` class.